### ISSUE TITLE
 Window Subsystem for Linux messes up mineflayer

### ISSUE THREAD

**Issue by EndermanHackerVN** [PrismarineJS/mineflayer#2413]
- [x] The [FAQ](https://github.com/PrismarineJS/mineflayer/blob/master/docs/FAQ.md) doesn't contain a resolution to my issue 

## Versions
 - mineflayer: lastest
 - server: 1.17.1 purpurmc, 1.8.9 
 - node: 17.13.1

## Detailed description of a problem
Mineflayer threw out econnrefused everytime it tries to connect to any localhost IP
When testing mineplex / any servers it threw out `SyntaxError: Unexpected token u in JSON at position 0`

![image](https://user-images.githubusercontent.com/67498765/150059891-e483ffc9-7745-4272-9fec-03b36b6c702e.png)

![image](https://user-images.githubusercontent.com/67498765/150060526-c3ae131c-7127-4366-9af5-1497a9bcb69e.png)

## What did you try yet?
Did you try any method from the API? Yes
Did you try any example? Any error from those? Yes

## Your current code
```js
const bot = mineflayer.createBot({
        host: 'us.mineplex.com', 
        username: user, 
        password: pass,
        port: 25565,            
        auth: 'microsoft',
        colorsEnabled: true,
        hideErrors: true
    })

    bot.on('spawn', () => {
        bot.chat('Hello world!')
    })
```

## Expected behavior
It should login.

## Additional context
N/A

**Comment by indaqoo** [comment]
I've just tested out with my premium account and I can login with premium account to hypixel, but I was not able to login mineplex...
![image](https://user-images.githubusercontent.com/28653209/150106644-6a11936a-2705-4b59-90f0-5a1174f09ad1.png)
![image](https://user-images.githubusercontent.com/28653209/150106686-14437422-40a6-4cee-943b-071f8aecb876.png)

I can see that you are using WSL.
Maybe that's network related since it cannot connect to microsoft's services since you are using "auth: 'microsoft'"

**Comment by extremeheat** [comment]
This has nothing to do with the environment, I think the error is pretty clear. Can’t deserialize JSON from an invalid string

**Comment by nickelpro** [comment]
The problem is that Mineflayer expects signs to always contain valid json for text lines, and I'm not sure that's true for all versions.

**Comment by EndermanHackerVN** [comment]
I guess the WSL and mineplex server is the problem here.

Just to clarify, Im going to test at tomorrow, it is 23pm local time now. Thanks for the replies, I will update this issue soon.

**Comment by nickelpro** [comment]
Neither is the issue, the problem is in Mineflayer. A useful thing would be to get a copy of the block entity data by logging the block entity that's causing the problem before the problematic code that's causing the crash.

**Comment by nickelpro** [comment]
What this looks like based on the information available in the bug report, is that instead of containing JSON for text lines this sign entity uses Unicode strings directly.

**Comment by EndermanHackerVN** [comment]
TLDR: Window Subsystem for Linux messes up my connection with mineflayer

@nickelpro @extremeheat @Wrath-cyber @indaqoo 

Thanks for helping me solve my problem. Turns out WSL messed up connections somehow.

Localhost server:
![image](https://user-images.githubusercontent.com/67498765/150449472-24181a64-5e3b-4edf-8477-475679025152.png)

mc.hypixel.net:
![image](https://user-images.githubusercontent.com/67498765/150449640-c59a60bf-5b9d-4dea-a0bb-34326b030562.png)

Mineplex however, still throws out error. I guess it is because unicode characters on signs that caused this exception?
And there is a message listener too, I guess I'll just add a null check before the console.log. Again, thanks for helping me.
![image](https://user-images.githubusercontent.com/67498765/150449804-df6718c3-6750-4ec4-b3c1-30e8694b6cde.png)

**Comment by extremeheat** [comment]
You do not get errors like this randomly. Data from high level minecraft packets cannot get corrupted in transit because then compression/encryption would fail. The fact that it fails on mineplex is indicative that some parsing code in mineflayer/prismarine package is not working correctly

**Comment by nickelpro** [comment]
I don't know how to say "this has nothing to do with WSL" any more clearly. This is, as the error very clearly says, a bug with how Mineflayer handles signs.
